tla-io/src/main/scala/at/forsyte/apalache/io/annotations/parser/AnnotationLexer.scala

```diff
@@ -95,7 +95,7 @@ object AnnotationLexer extends RegexParsers {
   }
 
   private def number: Parser[NUMBER] = {
-    "(|-)[0-9][0-9]*".r ^^ { value => NUMBER(value.toInt) }
+    "(|-)[0-9][0-9]*".r ^^ { value => NUMBER(BigInt(value)) }
   }
```

Comment:
Why do we parse out random numbers in comments at all? Shouldn't we only proceed with parsing annotations if they begin with the `@` indicator? @konnov

Reply:
That's a good point. In the earlier versions, we were looking for `@foo(...)`, but that somehow did not work with multiline strings that were starting with `\*`. So we needed a proper lexer. And now we have this issue that the lexer was not accepting a big integer. The challenge here is how to filter out an annotation from the sea of arbitrary characters.